data/text.py

```diff
@@ -61,14 +61,19 @@ def match_like(field, search_query):
     return Field.__pow__(field, clause)
 
 
-def regex_search(query, field, pattern, matches=True):
-    return query.where(field.regexp(pattern)) if matches else query.where(~field.regexp(pattern))
+def regex_search(query, field, pattern, offset, limit, matches=True):
+    return (
+        query.where(field.regexp(pattern)).offset(offset).limit(limit)
+        if matches
+        else query.where(~field.regexp(pattern)).offset(offset).limit(limit)
+    )
 
 
-def regex_sqlite(query, field, pattern, matches=True):
+def regex_sqlite(query, field, pattern, offset, limit, matches=True):
     rows = query.execute()
```

Comment:
Can we get a warning and/or a comment saying that the query is going to fetch all rows regardless of what offset and limit is set to?

Reply:
good call, added.